### PR TITLE
Refactoring and improvements

### PR DESCRIPTION
**Pull request by slacker-by-design**
- First commit improves logger configuration behaviour
- Second is repository module review and refactoring
- Third is controller module review and refactoring

### CODE REVIEW

**Review by sambogrub — APPROVED** [review]
I'll have to mess with this on my own a bit, but it does seem straight forward. I'll have to look up the .mkdir() method though.

**Review by sambogrub — APPROVED** [review]
It makes sense that the controller would move. 
Was there a specific reason that you implemented just datetime? 
since you suggested moving the connection to the __init__ as a class attribute, does that mean the only time to close the connection is when the app is closed?
I like separating the initializing of the tables, it creates all of the tables in that same function right?

**Comment by slacker-by-design** [comment]
>  Was there a specific reason that you implemented just datetime?
Name conflict. This `from datetime import date` doesn't work well with `def populate_month_data(self, date: date)`. Once you start using variable `date` and class `date` in the same code, it'll lead to a mess.

**Review by sambogrub — APPROVED** [review]

**Comment by slacker-by-design** [comment]
> I like separating the initializing of the tables, it creates all of the tables in that same function right?

Well, considering the "complexity" of your DB, one function will work just fine. It's almost always about considering the complexity of dealing with several smaller functions / classes vs one huge - do it all function / class. 

Former approach makes it easy to use individual functions / classes, but orchestration (e.g. keeping the right order of call) can be difficult. The latter can be difficult to use correctly due to amount of input parameters it may need to function properly.

Basically - just try to use "gut feeling" :-) 

P.S.
Having "DB structural staff" separated from "data CRUD stuff" can be even bigger advantage when you need to change the DB structure from time to time...
